### PR TITLE
ENT-4108 | added management command `populate_job_names`

### PR DESCRIPTION
**Pull request by HammadAhmadWaqas**
**Merge checklist:**
- [ ] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `make upgrade && make requirements` have been run to regenerate requirements
- [ ] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
- [ ] [Version](https://github.com/edx/taxonomy-connector/blob/master/taxonomy/__init__.py) bumped
- [ ] [Changelog](https://github.com/edx/taxonomy-connector/blob/master/CHANGELOG.rst) record added

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/taxonomy-connector/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [GitHub Actions](https://github.com/edx/taxonomy-connector/actions), verify version has been pushed to [PyPI](https://pypi.org/project/taxonomy-connector/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [course-discovery](https://github.com/edx/course-discovery) to upgrade dependencies (including taxonomy-connector)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in course-discovery will look for the latest version in PyPi.

### CODE REVIEW

File: taxonomy/emsi_client.py
Comment (sameenfatima78) [inline taxonomy/emsi_client.py:186]: Nit: fetch -> fetched

File: taxonomy/emsi_client.py
Comment (sameenfatima78) [inline taxonomy/emsi_client.py:190]: Can we change "details" to something more specific? Talking about the second "detail"

File: taxonomy/management/commands/populate_job_names.py
Comment (sameenfatima78) [inline taxonomy/management/commands/populate_job_names.py:23]: Nit: no -> not, Job -> jobs

File: taxonomy/management/commands/populate_job_names.py
Comment (sameenfatima78) [inline taxonomy/management/commands/populate_job_names.py:21]: Nit: Command*

File: taxonomy/management/commands/populate_job_names.py
Comment (sameenfatima78) [inline taxonomy/management/commands/populate_job_names.py:31]: I think we already have a helping method with a similar name which may add confusion, how about we make this one more specific and change it to `_update_job_name` instead?

File: taxonomy/management/commands/populate_job_names.py
Comment (sameenfatima78) [inline taxonomy/management/commands/populate_job_names.py:41]: Nit: 
"Updated the Jobs by fetching data from EMSI" 
->
"Fetch the Jobs with missing names from EMSI and update their names."

File: taxonomy/management/commands/populate_job_names.py
Comment (sameenfatima78) [inline taxonomy/management/commands/populate_job_names.py:63]: Can we use the same formatting standard everywhere to keep things consistent? Either use `f strings` or `format strings`.

File: taxonomy/management/commands/populate_job_names.py
Comment (sameenfatima78) [inline taxonomy/management/commands/populate_job_names.py:29]: Nit: Populates*

File: taxonomy/management/commands/populate_job_names.py
Comment (sameenfatima78) [inline taxonomy/management/commands/populate_job_names.py:71]: Would be nice if we can add some logging here when this command runs successfully

File: tests/test_emsi_client.py
Comment (sameenfatima78) [inline tests/test_emsi_client.py:257]: Nit: Remove "that"

File: tests/test_emsi_client.py
Comment (sameenfatima78) [inline tests/test_emsi_client.py:271]: Nit: Remove "that"

**Review by sameenfatima78 — APPROVED** [review]
Some modifications needed, otherwise LGTM

File: tests/test_emsi_client.py
Comment (sameenfatima78) [inline tests/test_emsi_client.py:274]: Indentation seems to be a bit out of place here

File: taxonomy/management/commands/populate_job_names.py
Comment (HammadAhmadWaqas) [reply taxonomy/management/commands/populate_job_names.py:71]: added

File: taxonomy/management/commands/populate_job_names.py
Comment (HammadAhmadWaqas) [reply taxonomy/management/commands/populate_job_names.py:62]: fixed the other, now using f strings

**Review by sameenfatima78 — APPROVED** [review]
LGTM (once build passes)

File: tests/test_emsi_client.py
Comment (HammadAhmadWaqas) [reply tests/test_emsi_client.py:274]: This is expected Indentation here. Otherwise, we get a quality error.

Comment (HammadAhmadWaqas) [reply tests/test_emsi_client.py:274]: (bad-continuation)

**Review by saleem-latif — APPROVED** [review]
Please use single-quoted strings e.g. `'test string "quoted"'` in all places, once this is done :+1:

File: taxonomy/emsi_client.py
Comment (muhammad-ammar) [inline taxonomy/emsi_client.py:183]: typo: loopup -> lookup

**Review by muhammad-ammar — APPROVED** [review]

**Pull request merged**
